extensions/metadata_exchange/plugin.cc

```diff
@@ -64,14 +64,6 @@ bool serializeToStringDeterministic(const google::protobuf::Message& metadata,
 static RegisterContextFactory register_MetadataExchange(
     CONTEXT_FACTORY(PluginContext), ROOT_FACTORY(PluginRootContext));
 
-static RegisterContextFactory register_StatsOutbound(
-    CONTEXT_FACTORY(PluginContext), ROOT_FACTORY(PluginRootContextOutbound),
-    "mx_outbound");
-
-static RegisterContextFactory register_StatsInbound(
```

Comment:
Thanks for catching that. The duplicate name is harmless since it's namespaced I think.

Reply:
Is static variable follows the namespace scope? This seems to be causing problem in the integration test. The test fails when specifying mx_inbound/mx_outbound as root context, complaining it cannot find the root context. I think it might be race between these and the same static variables in stats plugin.

Reply:
Might be some UB. Let's not try our luck and just clean up the names.

Reply:
Do we need two stats roots if we don't have a cache there?

Reply:
It uses the root context indication to distinguish listener direction at data path. It is not necessary, but need some code change and cares to make sure nothing breaks.

Reply:
It will be a breaking change someone is not using the default installation. We could remove the usage of root id in the code, as well as from default configuration, but might need to keep the declaration in code just for safety and remove it later with some announcement. So not a free clean up :)